Assets/Scripts/RocketScript.cs

```diff
@@ -9,6 +9,7 @@ public class RocketScript : MonoBehaviour, IRocket {
     private bool beginGame;
     private IInputManager inputManager;
     private IBase baseScript;
+    private bool landed;
```

Comment:
I was going to say this condition isn't necessary since we implicitly have this information in "rotationCenter" (whether or  not that's null), but I realized that that wouldn't work either since rotationCenter is null at game start and we still want to listen for input there.

That makes me think though that really what we want is a condition of _is the rocket allowed to launch_, which in this case we're extrapolating from _landed_ but that might not necessarily be true in the future. Even in the case of using _rotationCenter_ (which wouldn't work), we're still extrapolating out information. I feel like whether or not the overall game conditions permit the rocket to launch is out of scope of the rocket script's responsibilities.

@roropower What do you think of having this condition instead be "readyToLaunch" or something, and have a method that BaseScript can set this condition instead? I feel like that might make this behavior more robust in that we don't have to track all conditions that will allow us to launch within the rocket script and instead allow that logic to live more globally. Or is that too roundabout for what we're looking for here?

Reply:
No, that's perfect. I can fix that in the future.